### PR TITLE
Speedup bootstrap

### PR DESCRIPTION
**Pull request by ayushpatnaikgit**

### CODE REVIEW

**Comment by ayushpatnaikgit** [comment]
@iuliadmtru can you help? I haven't changed `show.jl` or it's test or any lines related to it.

File: docs/make.jl [hunk 1]
```diff
@@ -7,6 +7,7 @@ DocMeta.setdocmeta!(Survey, :DocTestSetup, :(using Survey); recursive=true)
 makedocs(;
     modules=[Survey],
     authors="xKDR Forum",
+    # doctest = :fix,
```
Comment (ayushpatnaikgit) [inline docs/make.jl:10]: I have intentionally left this comment. It's an imperfection in the package, but makes our life a lot simpler. @smishr @iuliadmtru

File: test/total.jl [hunk 2]
```diff
@@ -173,12 +173,6 @@ end
     @test filter(:cname => ==("Los Angeles"), tot).SE[1] ≈ 292840.83 rtol = SE_TOL
     @test filter(:cname => ==("San Diego"), tot).total[1] ≈ 1227596.71 rtol = STAT_TOL
     @test filter(:cname => ==("San Diego"), tot).SE[1] ≈ 860028.39 rtol = SE_TOL
-    mn = mean(:api00, :cname, clus1)
```
Comment (ayushpatnaikgit) [inline test/total.jl:176]: Mean tests were being done in total.

File: docs/make.jl (hunk 1 above)
Comment (smishr) [reply docs/make.jl:10]: Yes. Using this in make simply updates all the docstest outputs to new API.

File: src/bootstrap.jl [hunk 3]
```diff
@@ -20,32 +22,25 @@ replicates: 1000
...
-            psus = unique(substrata[!, design.cluster])
-            if length(psus) <= 1
-                stratified[h].whij .= 0 # hasn't been tested yet. 
-            end
-            nh = length(psus)
-            randinds = rand(rng, 1:(nh), (nh-1)) # Main bootstrap algo. Draw nh-1 out of nh, with replacement.  
-            rh = [(count(==(i), randinds)) for i in 1:nh] # main bootstrap algo. 
-            gdf = groupby(substrata, design.cluster)
-            for i in 1:nh
-                gdf[i].whij = repeat([rh[i]], nrow(gdf[i])) .* gdf[i][!,design.weights] .* (nh / (nh - 1))
```
Comment (smishr) [inline src/bootstrap.jl:37]: @ayushpatnaikgit can you explain in a post how you achieved this speedup? I think the issue was the gdf[i], which u solved using manual counters, sorting substrata by cluster. Also there is a `push!`, which is not the fastest operation in Julia?

**Pull request merged**